src/structlog/types.py

```diff
@@ -97,6 +97,30 @@
 """
 
 
+@runtime_checkable
+class ProcessorExceptionFormatter(Protocol):
```

Comment:
ugh OK now I'm realizing we've gotten ourselves into a mess naming-wise :-/

We've got three things ending in "Formatter":

- ProcessorExceptionFormatter, a protocol
- DictExceptionFormatter, something that formats exc_info into dicts
- ExceptionFormatter, something that replaces exc_info by exceptions (processed potentially by DictExceptionFormatter)

We need to improve this. 😅

My gut feeling tells me that actually _none_ of them should be called a formatter.

- DictExceptionFormatter doesn't format anything, it returns a list of dicts.
- same argument applies to ProcessorExceptionFormatter, simply by virtue of the existence of DictExceptionFormatter

I suspect Renderer would be a better name here, since we use it for JSONRenderer too? I would appreciate your thoughts here tho.

A Formatter-suffix would be a good name for a class that formats to text, as the old processor did, but that doesn't seem to have an own class.

I would postpone the discussion about the naming of ExceptionFormatter until we've nailed the other two down

Reply:
Just my first thoughts, need to sleep over them:

- `ExceptionFormatter` -> renders exceptions into something that can be printed by the last processor -> `ExceptionRenderer`
- `DictExceptionFromatter` -> transforms a traceback from a long string to dicts -> `ExceptionDictTransformer`
- `ProcessorExceptionFormatter` -> A protocol for exception transformers used by the exception renderer -> `ExceptionTransformer`

Reply:
that's one looooong nap ;)

I think I'm OK with those names unless your super-nap unveiled some space brains. :)

Reply:
No, several naps did not lead to better names. 

I will change the code to these names. Then we can get a feeling of how they feel. 👌